### PR TITLE
Build: Improve Windows dependency download resilience

### PR DESCRIPTION
**Pull request by hoffie**
**Short description of changes**
- CI: Fix caching of NSIS & ASIO installations
  Previously, we were caching the wrong directories for NSIS/ASIO. Therefore, the cache never hit and wasn't effective against download problems.

- Build: Windows: Log when re-using NSIS/ASIO directories
  This makes it easier to spot problems in the caching logic.

- Build: Retry SourceForge (NSIS) downloads for improved build stability
  Previously, a single download failure made the whole build fail. Now we try three times in a row after a slight delay.

- Build: windows/deploy_windows: Make spacing/case more consistent
  ... while we are at it.

CHANGELOG: SKIP

**Context: Fixes an issue?**
Fixes: #2775

**Does this change need documentation? What needs to be documented and how?**
No.

**Status of this Pull Request**
Ready.

**What is missing until this pull request can be merged?**
Reviews.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

Test results:
- [x] Cache size increased from [179MB (master)](https://github.com/jamulussoftware/jamulus/runs/7755249088?check_suite_focus=true#step:5:12) to [186MB (a stripped down version of this PR)](https://github.com/hoffie/jamulus/runs/7808199877?check_suite_focus=true#step:5:12)
- [x] The cache hit is [logged as expected now](https://github.com/hoffie/jamulus/runs/7808199877?check_suite_focus=true#step:9:22)

### CODE REVIEW

File: windows/deploy_windows.ps1 [hunk 1]
```diff
@@ -55,17 +55,30 @@ Function Clean-Build-Environment
...
-        [String]$URL
+        [string] $url
     )
 
-    $request = [System.Net.WebRequest]::Create($url)
-    $request.AllowAutoRedirect=$true
-    $response=$request.GetResponse()
-    $response.ResponseUri.AbsoluteUri
-    $response.Close()
+    foreach ($attempt in 1, 2, 3) {
```
Comment (ann0see) [inline windows/deploy_windows.ps1:63]: That’s a strange syntax. Is there any reason why that’s not a standard for loop (doesn’t that also exist in PS?)

File: .github/workflows/autobuild.yml [hunk 2]
```diff
@@ -213,8 +213,8 @@ jobs:
           path: |
             C:\Qt
             C:\ChocoCache
-            ~\windows\NSIS
-            ~\libs\ASIOSDK2
+            ${{ github.workspace }}\windows\NSIS
```
Comment (ann0see) [inline .github/workflows/autobuild.yml:216]: I probably didn’t take enough care about windows as I moved the SDK. Thanks for fixing

**Review by ann0see — APPROVED** [review]

File: windows/deploy_windows.ps1 (hunk 1 above)
Comment (hoffie) [reply windows/deploy_windows.ps1:63]: Probably depends where one comes from. Python is foreach-centric... ;)
The alternative would be:

```suggestion
    for ($attempt = 1; $attempt -le 3; $attempt++) {
```

Not sure if it's better though.

File: .github/workflows/autobuild.yml (hunk 2 above)
Comment (hoffie) [reply .github/workflows/autobuild.yml:216]: I'm not even sure what broke it, maybe it had never worked correctly, although I vaguely remember putting lots of effort into the testing. Anyway, now it should be good.

File: windows/deploy_windows.ps1 (hunk 1 above)
Comment (ann0see) [reply windows/deploy_windows.ps1:63]: I think it’s more standard, but it seems to be more verbose. 
I‘d go with the foreach loop then.

**Pull request merged**